Add style word array bracket to .rubocop.yml

marcushwz opened the pull request:
I am not sure if I am missing anything, but according to the style guide, literal array syntax is preferred over `%i` and `%w`.
```
# bad
STATES = %w(draft open closed)

# good
STATES = ["draft", "open", "closed"]
```

- However, looking at the current Cop enabled, [Style/WordArray](https://msp-greg.github.io/rubocop/RuboCop/Cop/Style/WordArray.html) is not included inside.  I have extended it manually for my project, so that I will get linting message like this:
![image](https://user-images.githubusercontent.com/40255418/115135798-d3b96280-a04d-11eb-8ad7-0b1dd40d51e5.png)

- Therefore, I am making this PR, to add in the `Style/WordArray` and set it to `bracket` as this is the preferred style as mentioned in the style guide.

volmer commented on rubocop.yml:
```diff
@@ -1027,3 +1027,7 @@ Lint/DeprecatedOpenSSLConstant:
+
+Style/WordArray:
+  EnforcedStyle: brackets
+  Enabled: true
```
You don't need this line. By declaring the cop with a config it is going to be enabled already.

volmer reviewed (approved):
Thanks for catching this!

volmer commented:
@marcushwz Could you please resolve the conflicts and squash your commits? I'll merge it afterwards.

volmer commented:
Also could you please enable `SymbolArray` as well?

marcushwz commented:
@volmer Done adding in SymbolArray with brackets enforced style. I have two questions:
1. Do we still want the `Lint/Syntax` cop to be enabled? I noticed it was removed, I have added it back in but let me know if we don't want it to be included.
2. How do I update the `Description` in `full_config.yml`? I was thinking to update the Description to something like `Use [] for arrays of words`. But doing so will fail the `config test`. Then when I ran, `be rake config:dump` the description reverts back to original.

volmer commented:
> Do we still want the Lint/Syntax cop to be enabled? I noticed it was removed, I have added it back in but let me know if we don't want it to be included.

Yes, Lint/Syntax is now enabled by default. You don't need to enable it explicitly. You can notice that `full_config.yml` already has it enabled.

> How do I update the Description in full_config.yml? 

Don't worry about the description. By changing the config RuboCop will output the proper offence message accordingly.

marcushwz commented:
> > Do we still want the Lint/Syntax cop to be enabled? I noticed it was removed, I have added it back in but let me know if we don't want it to be included.
> 
> Yes, Lint/Syntax is now enabled by default (see #220). You don't need to enable it explicitly. You can notice that `full_config.yml` already has it enabled.
> 
> > How do I update the Description in full_config.yml?
> 
> Don't worry about the description. By changing the config RuboCop will output the proper offence message accordingly.

Okay, I have removed the `Lint/Syntax`, should be fine now. Let me know if you need me to change anything else. Thanks and have a great day!

volmer commented:
Thank you @marcushwz !
